src/backend/distributed/planner/intermediate_result_pruning.c

```diff
@@ -228,6 +248,22 @@ FindAllWorkerNodesUsingSubplan(HTAB *intermediateResultsHash,
 		}
 	}
 
+	/* don't include the current worker if the result will be written to local file */
+	if (entry->writeLocalFile)
+	{
+		WorkerNode *workerNode = NULL;
+		int32 localGroupId = GetLocalGroupId();
+
+		foreach_ptr(workerNode, workerNodeList)
+		{
+			if (workerNode->groupId == localGroupId)
+			{
+				workerNodeList = list_delete(workerNodeList, workerNode);
```

Comment:
`list_delete` is for lists of nodes. We should use `list_delete_ptr()` here, as `WorkerNode` is not a postgres node structure.

Reply:
thanks for pointing this out.